catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/metacard/workspace/WorkspaceTransformerTest.java

```diff
@@ -15,116 +15,280 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
 import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
 import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
 import ddf.catalog.data.impl.BinaryContentImpl;
-import ddf.catalog.data.types.Core;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
 import ddf.catalog.transform.InputTransformer;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.charset.Charset;
 import java.util.Map;
+import javax.annotation.Nullable;
 import org.apache.commons.io.IOUtils;
+import org.codice.ddf.catalog.ui.metacard.workspace.transformer.WorkspaceKeyTransformation;
+import org.codice.ddf.catalog.ui.metacard.workspace.transformer.WorkspaceTransformer;
+import org.codice.ddf.catalog.ui.metacard.workspace.transformer.WorkspaceValueTransformation;
 import org.codice.ddf.catalog.ui.util.EndpointUtil;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.mockito.Mockito;
 
 public class WorkspaceTransformerTest {
+  private final String KEY_TRANSFORMATION_JSON_KEY = "jsonKey";
 
-  private CatalogFramework cf;
+  private final String KEY_TRANSFORMATION_METACARD_KEY = "metacardKey";
 
-  private InputTransformer it;
+  private final Serializable KEY_TRANSFORMATION_VALUE = 15.5;
 
-  private WorkspaceTransformer wt;
+  private final WorkspaceKeyTransformation KEY_TRANSFORMATION =
+      new WorkspaceKeyTransformation() {
+        @Override
+        public String getMetacardKey() {
+          return KEY_TRANSFORMATION_METACARD_KEY;
+        }
 
-  private EndpointUtil ut;
+        @Override
+        public String getJsonKey() {
+          return KEY_TRANSFORMATION_JSON_KEY;
+        }
+      };
+
+  private final String VALUE_TRANSFORMATION_KEY = "targetKey";
+
+  private final int VALUE_TRANSFORMATION_METACARD_VALUE = 10;
+
+  private final double VALUE_TRANSFORMATION_JSON_VALUE = 10.5;
+
+  private final WorkspaceValueTransformation<Integer, Double> VALUE_TRANSFORMATION =
```

Comment:
Any reason not to just mock these out?

Reply:
probably just a ✏️ , imo it's more readable in a mocked form

Reply:
I vote for mocking instead of an anonymous class.

Reply:
Okay. I can change it.

Reply:
I was wrong. As it turns out, mocking these is pretty problematic.

The main issue is that if I were to make them mocks, every single method on each would have to be implemented on the mock; this not only means more redundant code implementing the things like value transformation methods on the key transformations and such, but it also means that the default methods in interfaces like `WorkspaceKeyTransformation` would not be tested.

I tried looking into Mockito's Spy capabilities as well, but apparently interfaces can't be Spied.

If there are other suggested ways of doing it or I'm misunderstanding something, let me know and I'll try to accomodate.